### PR TITLE
[containerapp] Add Support for Qdrant dev service

### PR DESCRIPTION
**Pull request by bgashirabake**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
containerapp

**Comment by bgashirabake** [comment]
this is ready for review, i believe.

**Comment by bgashirabake** [comment]
more comment add.

**Comment by bgashirabake** [comment]
added env deletion after test run.

**Comment by bgashirabake** [comment]
@zhoxing-ms could you please help review this PR. it's a part of ongoing Springboard dev services work for container apps and some upcoming demos are contingent on it. thank you!!

**Comment by zhoxing-ms** [comment]
@Greedygre Could you please help review this PR?

**Comment by zhoxing-ms** [comment]
https://github.com/Azure/azure-cli-extensions/pull/6726#issuecomment-1707396143 If you want to release a new extension version for this PR, please also upgrade the version defined in `setup.py`

**Comment by Greedygre** [comment]
Hi @bgashirabake 
Could you please comments for fix the history.rst.

**Comment by Greedygre** [comment]
> @Greedygre Could you please help review this PR?

I have reviewed this PR.

**Comment by Greedygre** [comment]
> [#6726 (comment)](https://github.com/Azure/azure-cli-extensions/pull/6726#issuecomment-1707396143) If you want to release a new extension version for this PR, please also upgrade the version defined in `setup.py`

BTW, #bgashirabake 
We generally release new extension versions periodically. Please do not update `setup.py` .

**Comment by bgashirabake** [comment]
> @Greedygre Could you please help review this PR?

i hadn't seen @Greedygre's comments. resolving them now.

**Review by Greedygre — APPROVED** [review]

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ Release History
 upcoming
 ++++++
 * Add command group 'az containerapp connected-env', support show/list/delete/create connected environment
-
+* 'az containerapp service': add support for creation and deletion of Qdrant vector database as a container app dev service
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:8]: please add one blink line.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**